src/unread/UnreadCards.js

```diff
@@ -62,6 +68,15 @@ export default function UnreadCards(props: Props): Node {
     return <SearchEmptyState text="No unread messages" />;
   }
 
+  // This method return the number of unread mentions per topic
+  const getMentions = (stream_id: number, topic: string) => {
+    let unreadMentions = 0;
+    const streamData = unreadStreamsAndTopics.find(item => item.streamId === stream_id);
+    const topicData = streamData?.data.find(item => item.topic === topic);
```

Comment:
Searches like these are inefficient; see #3339. We have some in the existing code, but shouldn't add more.

Here, this is quite inefficient overall because for every one of the `TopicItem` components, we're going to end up searching through the whole list of unread streams and the whole list of unread topics in the relevant stream. So that's a loop of loops, and takes quadratic time.

Reply:
Thanks, I now realized the mistake I made. I was trying to categorize how many mentions each topic had.

So before I make the changes, I just want to understand the suggested [implementation](https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/unread.20mentions/near/1337465). In this, it is mentioned to compare the unread mentions with the local msgs.  So when we return the data of the topic, we can also return a boolean (if the user is mentioned or not) after comparing the msgIds with the unread mentions. To return it, we can use the data attribute in the variable named `total`.Does this sound right?

https://github.com/zulip/zulip-mobile/blob/255e1b49e516045403871cfef2b9251b0d51aefb/src/unread/UnreadCards.js#L29-L47

Reply:
> So when we return the data of the topic, we can also return a boolean (if the user is mentioned or not) after comparing the msgIds with the unread mentions. To return it, we can use the data attribute in the variable named `total`.Does this sound right?

Yep, that's the right spot.


> Thanks, I now realized the mistake I made. I was trying to categorize how many mentions each topic had.

This is actually unrelated to the inefficiency here. If we did want to count the number of mentions, we would still want to do so inside the `getUnreadStreamsAndTopics` selector (so that the computation doesn't get repeated if the component rerenders for some unrelated reason), and would still want to do so in a way that looked at each piece of data just once so that it runs in linear time (aka O(n) time.)

For example, your current draft has:
```js
        let isMentioned = false;
        msgIds.forEach(item => {
          if (unreadMsgIds.has(item)) {
            isMentioned = true;
          }
        });
```

If we wanted the number of mentions instead of a boolean saying whether there are any mentions, we could change that to:
```js
        let numMentioned = 0;
        msgIds.forEach(item => {
          if (unreadMsgIds.has(item)) {
            numMentioned += 1;
          }
        });
```
and that would be just as efficient as the current draft.